gql_websocket_link/test/gql_websocket_link_test.dart

```diff
@@ -1,54 +1,360 @@
-// Adapted to `gql` by @iscriptology
-
-import "dart:async";
 import "dart:convert";
+import "dart:io";
 
-import "package:gql_exec/gql_exec.dart";
 import "package:gql/language.dart";
-import "package:gql_websocket_link/src/messages.dart";
-import "package:gql_websocket_link/src/socket_client.dart";
+import "package:gql_exec/gql_exec.dart";
 import "package:gql_websocket_link/gql_websocket_link.dart";
-import "package:gql_link/gql_link.dart";
-import "package:mockito/mockito.dart";
+import "package:web_socket_channel/io.dart";
 import "package:test/test.dart";
 
-class MockClient extends Mock implements SocketClient {}
-
-class MockRequestSerializer extends Mock implements RequestSerializer {}
+void main() {
+  group(
+    "WSLink",
+    () {
+      test(
+        "send connection_init",
+        () async {
+          HttpServer server;
+          WebSocket webSocket;
+          IOWebSocketChannel channel;
+          WSLink link;
+          Request request;
 
-class MockResponseParser extends Mock implements ResponseParser {}
+          request = Request(
+            operation: Operation(
+              operationName: "sub",
+              document: parseString("subscription MySubscription {}"),
+            ),
+          );
 
-void main() {
-  group("WSLink", () {
-    MockClient client;
-    Request request;
-    WSLink link;
+          server = await HttpServer.bind("localhost", 0);
+          server.transform(WebSocketTransformer()).listen(
+            (webSocket) async {
+              final channel = IOWebSocketChannel(webSocket);
+              channel.stream.take(1).listen(
+                expectAsync1<void, dynamic>(
+                  (dynamic message) {
+                    final map =
+                        json.decode(message as String) as Map<String, dynamic>;
+                    expect(map["type"], MessageTypes.connectionInit);
+                  },
+                ),
+              );
+            },
+          );
 
-    final Stream<Response> Function([
-      Request customRequest,
-    ]) execute = ([
-      Request customRequest,
-    ]) =>
-        link.request(customRequest ?? request);
+          webSocket = await WebSocket.connect("ws://localhost:${server.port}");
+          channel = IOWebSocketChannel(webSocket);
 
-    setUp(() {
-      client = MockClient();
-      request = Request(
-        operation: Operation(
-          document: parseString("subscription MyQuery {}"),
-        ),
-        variables: const <String, dynamic>{"i": 12},
-      );
-      link = WSLink(
-        "/graphql-test",
+          link = WSLink(channel: channel);
+          //
+          link.request(request).listen(print);
+        },
       );
-    });
-  });
 
-  /*
-  * For now no tests present because the [SocketClient] class has its own tests,
-  * nonetheless tests here would be super important to ensure the quality of the implementation
-  * Feel free to PR :)
-   */
+//      test(
```

Comment:
I would prefer not leaving in commented tests.

Reply:
tests will be fixed in next commit, please leave this PR open until all tests are passed.